school/lms/doctype/cohort/cohort.json

```diff
@@ -0,0 +1,142 @@
+{
+ "actions": [],
+ "allow_rename": 1,
+ "autoname": "format:{course}/{slug}",
+ "creation": "2021-11-19 11:45:31.016097",
+ "doctype": "DocType",
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "course",
+  "slug",
+  "title",
+  "section_break_2",
+  "instructor",
+  "status",
+  "column_break_4",
+  "begin_date",
+  "end_date",
+  "duration",
+  "section_break_8",
+  "description",
+  "pages"
+ ],
+ "fields": [
+  {
+   "fieldname": "description",
+   "fieldtype": "Markdown Editor",
+   "label": "Description"
+  },
+  {
+   "fieldname": "instructor",
+   "fieldtype": "Link",
+   "label": "Instructor",
+   "options": "User",
+   "reqd": 1
+  },
+  {
+   "fieldname": "status",
+   "fieldtype": "Select",
+   "label": "Status",
+   "options": "Upcoming\nLive\nCompleted\nCancelled",
+   "reqd": 1
+  },
+  {
+   "fieldname": "begin_date",
+   "fieldtype": "Date",
+   "label": "Begin Date"
+  },
+  {
+   "fieldname": "end_date",
+   "fieldtype": "Date",
+   "label": "End Date"
+  },
+  {
+   "fieldname": "duration",
+   "fieldtype": "Data",
+   "label": "Duration"
+  },
+  {
+   "fieldname": "slug",
+   "fieldtype": "Data",
+   "in_list_view": 1,
+   "label": "Slug",
```

Comment:
Why are we using slug when we can use a series? To a new user, this field won't make much sense as to what data is expected here. Same for slug in cohort subgroup.

Reply:
Slug is part of the URL. I would like to URLs to be meaningful.  For example, this is URL that we use now.

https://mon.school/courses/the-joy-of-programming/cohorts/th-2021-12

We could auto create the slug from not title, if not specified, but I would like to have an option to specify it.

Same with subgroups. Using names with case variations and spaces doesn't play well with URL and things break when we copy the URL and share it in email. Again, we could auto generate the slug from title.

Reply:
> but I would like to have an option to specify it

Why so when we can auto-create the slug from the title as you mentioned.

Reply:
For the course we don't have slug because, the name itself is the slug and the name could be edited. Where as for subgroup (and also for cohort) the name can not be derived from the title along as the same college can be part of two cohorts, so we use the autogenerated name and use a slug. The combination  of course, cohort-slug and subgroup-slug forms the unique URL.